app/utils/auth/providers.py

```diff
@@ -196,3 +196,21 @@ def get_userinfo(cls, user: models_core.CoreUser):
             "email": user.email,
             "groups": [group.name for group in user.groups],
         }
+
+
+class MinecraftAuthClient(BaseAuthClient):
+    # If no redirect_uri are hardcoded, the client will need to provide one in its request
+    redirect_uri: str | None = None
+    # Set of scopes the auth client is authorized to grant when issuing an access token.
+    # See app.utils.types.scopes_type.ScopeType for possible values
+    allowed_scopes: Set[ScopeType] = {ScopeType.profile}
+
+    @classmethod
+    def get_userinfo(cls, user: models_core.CoreUser):
+
+        return {
+            "id": user.id,
+            "nickname": user.nickname,
```

Comment:
A user may not have a surname. The firstname may be used in this case

Reply:
maybe it would be better to use Firstname + Lastname knowing that the surname will become your in-game name